koku/masu/database/ocp_report_db_accessor.py

```diff
@@ -283,6 +283,26 @@ def delete_hive_partitions_by_source(self, table, partition_column, provider_uui
         LOG.info(log_json(msg="successfully deleted Hive partitions", context=ctx))
         return True
 
+    def find_expired_trino_partitions(self, table, source_column, date_str):
+        """Queries Trino for partitions less than the parition date."""
+        if not self.table_exists_trino(table):
+            LOG.info("Could not find table.")
+            return False
+        sql = f"""
+SELECT partitions.year, partitions.month, partitions.source
+FROM (
+    SELECT year as year,
+        month as month,
+        day as day,
+        cast(date_parse(concat(year, '-', month, '-', day), '%Y-%m-%d') as date) as partition_date,
+        {source_column} as source
+    FROM  "{table}$partitions"
+) as partitions
+WHERE partitions.partition_date < DATE '{date_str}'
+GROUP BY partitions.year, partitions.month, partitions.source
+"""
```

Comment:
So, I did look into this and tried to utilize the built in `sql_params` available in the `_execute_trino_raw_sql_query`. However, when the params where passed to the EXECUTE IMMEDIATE query, it does not properly replace the `?` with the parameters.
```
EXECUTE IMMEDIATE '
SELECT partitions.year, partitions.month, partitions.source
FROM (
    SELECT year as year,
        month as month,
        day as day,
        cast(date_parse(concat(year, ''-'', month, ''-'', day), ''%Y-%m-%d'') as date) as partition_date,
        ? as source
    FROM  "?$partitions"
) as partitions
WHERE partitions.partition_date < DATE ''?''
GROUP BY partitions.year, partitions.month, partitions.source' USING 'source','reporting_ocpusagelineitem_daily_summary','2024-03-01'
```
Error: ` Incorrect number of parameters: expected 1 but found 3`
I did attempt to adjust my query to get something functional. I discovered a fix for the date. However, I am still running into issues trying to replace `"{table}$partitions"` within Trino itself. I even tried passing the entire from clause in as param to work around the double quotes. However, Trino does not seem happy with `?` in a from clause. 

```
trino:org1234567> EXECUTE IMMEDIATE '
               -> SELECT partitions.year, partitions.month, partitions.source
               -> FROM (
               ->     SELECT year as year,
               ->         month as month,
               ->         day as day,
               ->         cast(date_parse(concat(year, ''-'', month, ''-'', day), ''%Y-%m-%d'') as date) as partition_date,
               ->         ? as source
               ->     FROM  ?
               -> ) as partitions
               -> WHERE partitions.partition_date < CAST(? as DATE)
               -> GROUP BY partitions.year, partitions.month, partitions.source' USING 'source','"reporting_ocpusagelineitem_daily_summary$partitions"','2024-03-01';
```
I believe the solution to be to not use the `qmark` param style when preparing the query. However, that  would involve allowing me to change the default preprocessor for jinja queries in the `report_db_accessor_base.py`. At this point I decided maybe we are are starting to reach outside the scope of this issue. 

My vote is to create a tech debt issue for using parameters for all of our db accessor. I took a look and it appears we use f-strings for all of db_accessor files including the base. I think we should consider refactoring these as a whole.

Reply:
It's definintely not something that is easy to address, especially for all cases. A separate issue to address that is warranted. I want to call it out so we don't get accustomed to using string interpolation for SQL queries because it will bite us eventually.